elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -458,28 +462,34 @@ private JsonObject addFieldMapping(JsonObject payload, FacetMetadata facetMetada
 	}
 
 	private String getIndex(EntityIndexBinding binding, DocumentFieldMetadata fieldMetadata) {
-		// Never analyze boolean, date or numeric
+		String index = getIndexValue( fieldMetadata.getIndex() );
+
+		// Never analyze booleans, numerics, dates
 		if ( FieldHelper.isBoolean( binding, fieldMetadata.getName() ) ||
 				FieldHelper.isDate( binding, fieldMetadata.getName() ) ||
 				FieldHelper.isCalendar( binding, fieldMetadata.getName() ) ||
 				FieldHelper.isNumeric(fieldMetadata) ) {
-			return NOT_ANALYZED;
+			if ( ANALYZED.equals( index ) ) {
+				return NOT_ANALYZED;
+			}
 		}
 
-		return getIndexValue( fieldMetadata.getIndex() );
+		return index;
 	}
 
-	@SuppressWarnings("deprecation")
 	private String getIndex(BridgeDefinedField bridgeDefinedField) {
-		// Never analyze boolean
+		String index = getIndexValue( bridgeDefinedField.getIndex() );
+
+		// Never analyze booleans, numerics, dates
 		if ( FieldHelper.isBoolean( bridgeDefinedField )
 				|| FieldHelper.isNumeric( bridgeDefinedField )
 				|| FieldHelper.isDate( bridgeDefinedField ) ) {
-			return NOT_ANALYZED;
+			if ( ANALYZED.equals( index ) ) {
+				return NOT_ANALYZED;
+			}
 		}
 
-		Field.Index index = bridgeDefinedField.getIndex();
-		return getIndexValue( index );
+		return index;
 	}
 
 	private String getIndexValue(Field.Index index) {
```

Comment:
Right! I just amended the last commit, implementing what you suggested.

I changed it a bit, though: in order to avoid having to write `field.addProperty("xx", indexOptions.getXX())` multiple times in the code, I didn't use a IndexOptions structure, but rather transformed the `getIndexValue` method into a `addIndexOptions` method. I can't explain it well, but the code should be a bit clearer (I hope).

Also, I used this opportunity to simplify the "canBeAnalyzed" condition: if you look at `getFieldType()`, you'll see that the previous condition was listing every type except "string". So comparing the type to "string" is simpler.